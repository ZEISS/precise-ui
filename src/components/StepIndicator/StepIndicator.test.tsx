import * as React from 'react';
import * as enzyme from 'enzyme';
import { StepIndicator } from './';

describe('<StepIndicator />', () => {
  it('should render an empty step indicator', () => {
    const wrapper = enzyme.shallow(<StepIndicator steps={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a step indicator with two elements', () => {
    const wrapper = enzyme.shallow(<StepIndicator steps={['One', 'Two']} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a step indicator with three elements currently at the second', () => {
    const wrapper = enzyme.shallow(<StepIndicator steps={['One', 'Two', 'Three']} current={1} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a step indicator with two elements currently at the first', () => {
    const wrapper = enzyme.shallow(<StepIndicator steps={['One', 'Twee']} current={0} />);
    expect(wrapper).toMatchSnapshot();
  });
});
