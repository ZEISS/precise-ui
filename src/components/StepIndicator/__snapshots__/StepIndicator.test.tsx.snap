// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepIndicator /> should render a step indicator with three elements currently at the second 1`] = `
<styled.ol>
  <styled.li
    className=""
    key="0"
  >
    <styled.div>
      One
    </styled.div>
  </styled.li>
  <styled.li
    className="step-active"
    key="1"
  >
    <styled.div>
      Two
    </styled.div>
  </styled.li>
  <styled.li
    className=""
    key="2"
  >
    <styled.div>
      Three
    </styled.div>
  </styled.li>
</styled.ol>
`;

exports[`<StepIndicator /> should render a step indicator with two elements 1`] = `
<styled.ol>
  <styled.li
    className=""
    key="0"
  >
    <styled.div>
      One
    </styled.div>
  </styled.li>
  <styled.li
    className=""
    key="1"
  >
    <styled.div>
      Two
    </styled.div>
  </styled.li>
</styled.ol>
`;

exports[`<StepIndicator /> should render a step indicator with two elements currently at the first 1`] = `
<styled.ol>
  <styled.li
    className="step-active"
    key="0"
  >
    <styled.div>
      One
    </styled.div>
  </styled.li>
  <styled.li
    className=""
    key="1"
  >
    <styled.div>
      Twee
    </styled.div>
  </styled.li>
</styled.ol>
`;

exports[`<StepIndicator /> should render an empty step indicator 1`] = `<styled.ol />`;
